Features/CaseSearch/test_pages/casesearch_page.py

```diff
@@ -10,14 +16,157 @@ class CaseSearchWorkflows(BasePage):
+
+    def search_against_property(self, search_property, input_value, property_type, include_blanks=None):
+        if property_type == "TEXT_INPUT":
+            self.search_property = self.get_element(self.search_against_text_property_format, search_property)
+            self.wait_to_clear_and_send_keys(self.search_property, input_value)
+            self.send_keys(self.search_property, Keys.TAB)
+        elif property_type == "COMBOBOX":
+            self.combox_select_element = self.get_element(self.combox_select, search_property)
+            self.select_by_text(self.combox_select_element, input_value)
+            time.sleep(3)
+        if include_blanks == "YES":
+            self.select_include_blanks(search_property)
+
+    def date_range(self, no_of_days):
+        today_date = datetime.today()
+        sixty_days_ago = today_date - relativedelta(days=no_of_days)
+        date_ranges = str(sixty_days_ago.date()) + " to " + str(today_date.date())
+        return date_ranges
+
+    def check_help_text(self, search_property, help_text):
+        help_text = (By.XPATH, self.help_text_format.format(search_property, help_text))
+        assert self.is_visible_and_displayed(help_text)
+
+    def check_date_range(self, date_range):
+        date = self.get_element(self.date_selected, date_range)
+        print(date)
+        assert self.is_present(date)
+
+    def add_address(self, address, search_property):
+        address_search = self.get_element(self.search_for_address, search_property)
+        self.send_keys(address_search, address)
+        time.sleep(5)
+        self.send_keys(address_search, Keys.TAB)
+        time.sleep(10)
+
+    def check_json_function(self, city_address, type="HOME"):
+        if type == "HOME":
+            city_home = self.get_element(self.city_value_home, city_address)
+            assert self.is_present(city_home)
+        if type == "WORK":
+            city_work = self.get_element(self.city_value_work, city_address)
+            print(city_work)
+            assert self.is_present(city_work)
+
+    def check_search_screen_title(self, title):
+        title_on_screen = self.get_element(self.search_screen_title, title)
+        assert self.is_displayed(title_on_screen)
+
+    def assert_address_is_hidden(self, hidden_property):
+        hidden = self.get_element(self.search_property_name_combobox, hidden_property)
+        assert not self.is_displayed(hidden)
+
+    def select_include_blanks(self, search_property):
+        checkbox = self.get_element(self.include_blanks, search_property)
+        time.sleep(2)
+        self.js_click(checkbox)
+
+    def check_validations_on_property(self, search_property, property_type, message=None, required_or_validated="YES"):
+        validation_message_on_top = self.get_element(self.required_validation_on_top, search_property)
+        validation_message_per_prop = None
+        if property_type == "TEXT_INPUT":
```

Comment:
If the `property_type` like `TEXT_INPUT`, `COMBOBOX`, and even `YES` and `NO` are constants and the text would not change then I would suggest moving all of them to the constants file and importing them from there.

This can be done like
```py
# Features/CaseSearch/constants.py
TEXT_INPUT = "TEXT_INPUT"
YES="YES"
# ...and others
```

and they can be imported in this file and used here as

```py
from Features.CaseSearch.constants import TEXT_INPUT, YES, NO...
"""
all other tests
.
. 
"""
if property_type == TEXT_INPUT
``

This reduces the chances of typos and makes it easier to update if things change in the future.